src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/Interop/VsHierarchyExtensions.cs

```diff
@@ -28,39 +27,33 @@ public static Guid GetGuidProperty(this IVsHierarchy hierarchy, VsHierarchyPropI
         /// <summary>
         ///     Gets the value of the specified property if the hierarchy supports it, or throws an exception if there was an error.
         /// </summary>
-        [return: MaybeNull]
-        public static T GetProperty<T>(this IVsHierarchy hierarchy, VsHierarchyPropID property, T defaultValue = default)
+        public static T? GetProperty<T>(this IVsHierarchy hierarchy, VsHierarchyPropID property, T? defaultValue = default)
```

Comment:
Is the return only null if `defaultValue` is null? i.e. if I pass a non-null default, can I still get a null. If not, this can be annotated with `[return: NotNullIfNotNull("defaultValue")]`.

Same comment regarding the three changes further down in this file (for `out` parameters in two cases).

Reply:
No, it can also return `null` if the underlying call to `IVsHierarchy.GetProperty` returns an HRESULT indicating a failure.

Reply:
That doesn't appear to be true of the call below this one, where `Verify.HResult` presumably throws. Is this call not just delegating to that one? If so, perhaps the annotation can be improved as suggested.

Can `IVsHierarchy.GetProperty` produce a null `resultObject`? If so, then yeah there's nothing we can do to improve this.